### PR TITLE
Fixes to the asio/cereal stack

### PR DESCRIPTION
**Pull request by beckerhe**
This PR bundles some improvements which came out of the fuzzing effort.

1. The first commit frees a message after it was processed. This avoided a leak reported by the fuzztest. Probably this is not relevant in production because we never throw and recover at this point.
2. The second commit is a rewrite of the MessageOwner-class. Buffers allocated by asio are now given to a MessageOwner object which is then moved into the callback. This avoids a copy and maybe an allocation. This change came out of a segmentation fault triggered by the fuzzer. Like before I'm not entirely sure if that can happen in production, but I believe it's a good refactorting anyway.
3. Commit 3 makes applies some improvements to the TCP-handling-code. Fewer allocations, more stack-based objects, etc. I made these changes while I was chasing the segfault from commit 2. So no bug- or vulnerability-fix in here, at least none that I know of. But it cleans up the code a bit.

PS: Currently I don't consider merging this for 1.49. (I'm a bit afraid that these TCP changes might break something somewhere that I haven't tested so far). Please let me know if you think otherwise.

### CODE REVIEW

**Comment by dimitry-** [comment]
(I think this can go to 1.49)

File: OrbitCore/Message.h [hunk 1]
```diff
@@ -142,17 +141,25 @@ class Message {
...
 class MessageOwner : public Message {
  public:
-  MessageOwner(Message a_Message) {
-    Message* message = this;
-    *message = a_Message;
-    m_OwnedData.resize(m_Size);
-    memcpy(m_OwnedData.data(), m_Data, m_Size);
+  explicit MessageOwner(Message a_Message)
+      : Message(a_Message), m_OwnedData(m_Size) {
+    std::memcpy(m_OwnedData.data(), m_Data, m_Size);
```
[the remark this answers is not in the document]
Comment (beckerhe) [reply OrbitCore/Message.h:146]: Good point. Will change that!

[the remark this answers is not in the document]
Comment (beckerhe) [reply OrbitCore/Message.h:146]: ...by removing this overload entirely. It's not called.

**Comment by beckerhe** [comment]
Thanks to you all. I thought a bit more about the changes and decided to target 1.49.

**Pull request merged**